src/track/playcounter.cpp

```diff
@@ -0,0 +1,22 @@
+#include "track/playcounter.h"
+
+
+void PlayCounter::updatePlayed(bool bPlayed) {
+    if (bPlayed) {
+        // increment the play count when play during the current session
+        ++m_iTimesPlayed;
+        m_bPlayed = true;
+    } else if (m_bPlayed) {
+        // decrement the play count, but only if already played during
+        // the current session
+        if (0 < m_iTimesPlayed) {
```

Comment:
PlayCounter cannot guarantee consistency, because it has (and still needs) 
functions set all members individually. Therefore we need this check here. 
Otherwise the TrackDAO code has to be rewritten for PlayCounter.

On 11/12/2015 09:17 PM, Daniel Schürmann wrote:
>
> In src/track/playcounter.cpp 
> <https://github.com/mixxxdj/mixxx/pull/772#discussion_r44708692>:
>
> > @@ -0,0 +1,22 @@
> > +#include "track/playcounter.h"
> > +
> > +
> > +void PlayCounter::updatePlayed(bool bPlayed) {
> > +    if (bPlayed) {
> > +        // increment the play count when play during the current session
> > +        ++m_iTimesPlayed;
> > +        m_bPlayed = true;
> > +    } else if (m_bPlayed) {
> > +        // decrement the play count, but only if already played during
> > +        // the current session
> > +        if (0 < m_iTimesPlayed) {
>
> This condition should never be false.
> If it is false we had an inconsistent state before. Played checked but 
> play counter at 0.
> Lets fix this instead, if that still can happen.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/772/files#r44708692>.
>